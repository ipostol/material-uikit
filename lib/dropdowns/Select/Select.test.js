'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _Option = require('../Option');

var _Option2 = _interopRequireDefault(_Option);

var _Select = require('./Select');

var _Select2 = _interopRequireDefault(_Select);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('Select component', function () {

  it('should render correctly', function () {

    var props = {
      options: [{ id: 1, value: 'Option 1' }]
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Select2.default, props));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly value props', function () {

    var props = {
      options: [{ id: 1, value: 'Option 1' }],
      value: 1
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Select2.default, props));

    wrapper.find(_Option2.default).simulate('click');

    expect(wrapper).toMatchSnapshot();
  });
});