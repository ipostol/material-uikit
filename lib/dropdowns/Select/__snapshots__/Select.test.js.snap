// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render correctly 1`] = `
<div>
  <Label />
  <Dropdown
    className="fullWidth"
    contentWidth="100%"
    trigger={
      Array [
        <div
          className="trigger general placeholder"
      >
          Please select
      </div>,
      ]
    }
  >
    <div
      className="content"
    >
      <Option
        element="div"
        onClick={[Function]}
        size="l"
      >
        Option 1
      </Option>
    </div>
  </Dropdown>
</div>
`;

exports[`Select component should render correctly value props 1`] = `
<div>
  <Label />
  <Dropdown
    className="fullWidth"
    contentWidth="100%"
    trigger={
      Array [
        <div
          className="trigger general"
      >
          Option 1
      </div>,
      ]
    }
  >
    <div
      className="content"
    >
      <Option
        element="div"
        onClick={[Function]}
        size="l"
      >
        Option 1
      </Option>
    </div>
  </Dropdown>
</div>
`;
